### ISSUE TITLE
[suffix-subjects] Subjects with dollar sign ending don't cause an error

### ISSUE THREAD

**Issue by JasonWeinzierl** [JasonWeinzierl/eslint-plugin-rxjs-x#109]
The `suffix-subjects` rule isn't effecting a failure for the following code:

```ts
import { Subject } from 'rxjs';

const sourceSubject$ = new Subject<number>();
// Should cause an error because the suffix should be Subject, not Subject$.
```

It's possible this is intentional, to be compatible with `finnish`, but I can't tell for sure looking over the history of this rule.

Possibly relevant history:

- Initial implementation: https://github.com/cartant/eslint-plugin-rxjs/commit/e185ed8cabd2664c55cfbe3919ad868c2e760590
- Bug fixes for `$$` suffix: https://github.com/cartant/eslint-plugin-rxjs/issues/88
- 4.0.3 change: https://github.com/cartant/eslint-plugin-rxjs/commit/a23a69c0af965d5a1c7ac96879f9028eacfcbf26
- 5.0.2 change: https://github.com/cartant/eslint-plugin-rxjs/commit/d4a94c093dc1b05f066f95b86626872ddaf27e5d

**Comment by JasonWeinzierl** [comment]
After taking a closer look at the unit tests (and the older tslint implementation https://github.com/cartant/rxjs-tslint-rules/pull/77), it looks like this is intentional behavior.  Instead of changing behavior, we will explicitly document this behavior in the `suffix-subjects` documentation.  See #91 for documentation changes.
